packages/network-controller/src/NetworkController.ts

```diff
@@ -293,11 +295,22 @@ export class NetworkController extends BaseControllerV2<
         this.#setupInfuraProvider(type);
         break;
       case NetworkType.localhost:
-        this.#setupStandardProvider(LOCALHOST_RPC_URL);
+        if (chainId === undefined) {
```

Comment:
The checks for `chainId` were added because we need to convert `chainId` from a decimal number to a hexadecimal number (the provider config object in this version of NetworkController stores a `chainId` in decimal, but `createNetworkClient` takes a `chainId` in hex in order to create a static middleware for `eth_chainId`). This requires that `chainId` be something.